bftengine/include/bftengine/DbCheckpointManager.hpp

```diff
@@ -0,0 +1,55 @@
+// Concord
+//
+// Copyright (c) 2021 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the
+// "License").  You may not use this product except in compliance with the
+// Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include "Serializable.h"
+
+#include <vector>
+#include <chrono>
+#include <PrimitiveTypes.hpp>
+#include "InternalBFTClient.hpp"
+
+namespace bftEngine::impl {
+using SeqNum = bftEngine::impl::SeqNum;
+using InternalBftClient = bftEngine::impl::InternalBFTClient;
+class DbCheckpointManager {
+ public:
+  void sendInternalCreateDbCheckpointMsg(const SeqNum& seqNum, bool noop);
+  void enableDbCheckpoint(bool enable) { enableDbCheckpoint_ = enable; }
```

Comment:
FMPOV: It is better to make parameters as `const &` for class objects, structures, string, and other user-defined types only. Primitive datatypes should be always passed as `const`. Reason: It is not so useful to access the primitive types via references because it involves extra work to read from the earlier addresses. Whereas, for user-defined objects, passing `const &` saves stack. Wdyt?